variables.tf

```diff
@@ -14,13 +14,13 @@ variable "name_suffix" {
 variable "ip_ranges" {
   description = "A map of IP CIDR ranges (including their /x parts) that should be used by the public/private subnets for the various components of the infrastructure. See comments in source code for elaboration on accepted keys and suggested IP CIDR ranges. Can use an IP calculator (like https://www.calculator.net/ip-subnet-calculator.html) for help with calculating subnets & IP ranges."
   type = object({
-    public             = list(string)                                   # list of CIDR ranges - each with their /x parts (/20 advised) for public subnets of the VPC.
-    private_primary    = string                                         # a CIDR range including /x part (/20 advised) for primary IPs in private subnet of the VPC.
-    private_k8s        = list(object({ pods = string, svcs = string })) # list of objects of CIDR ranges - each with their /x parts (/20 advised) - for pods & services in a k8s cluster.
-    private_redis      = list(string)                                   # list of CIDR ranges - each with their /x parts (/29 advised) - for Redis. See https://www.terraform.io/docs/providers/google/r/redis_instance.html#reserved_ip_range
-    private_g_services = string                                         # a CIDR range including /x part (/20 advised) for Google services producers (like CloudSQL, Firebase, etc) in private subnet of the VPC.
-    proxy_only         = string                                         # an empty string or a CIDR range including /x part (/24 advised) for Proxy-Only subnet. Use empty string "" to avoid creating Proxy-Only subnet. See https://cloud.google.com/load-balancing/docs/l7-internal/proxy-only-subnets#proxy_only_subnet_create
-    serverless_access  = list(string)                                   # list of CIDR ranges - each with their /x parts (/28 advised) for Serverless VPC Access. See https://www.terraform.io/docs/providers/google/r/vpc_access_connector.html#ip_cidr_range
+    public             = list(string)                                                  # list of CIDR ranges - each with their /x parts (/20 advised) for public subnets of the VPC.
+    private_primary    = string                                                        # a CIDR range including /x part (/20 advised) for primary IPs in private subnet of the VPC.
+    private_k8s        = list(object({ name = string, pods = string, svcs = string })) # list of objects of CIDR ranges - each with their /x parts (/20 advised) - for pods & services in a k8s cluster.
```

Comment:
Since a name key is introduced, It's no longer backward compatible. Probably need to add a migration update in the ReadMe that this Version required `name` to be pass as empty `name = ""` to retain the old resources.

Reply:
Sure a README would help. But, that much we can still figure out from the produced plan. By non-breaking change i meant at least there is a way to handle this without having to go through destruction / recreation of resources